addons/l10n_id/migrations/1.2/end-migrate_update_taxes.py

```
@@ -0,0 +1,41 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, SUPERUSER_ID
+from odoo.tools import float_compare
+
+
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+
+    # create tax groups in format of [(xml_id, name)]
+    tax_group_info = [
+        ("l10n_id_tax_group_non_luxury_goods", "Non-luxury Good Taxes"),
+        ("l10n_id_tax_group_0", "Zero-rated Taxes"),
+        ("l10n_id_tax_group_exempt", "Tax Exempted"),
+    ]
+
+    for xmlid, name in tax_group_info:
+        if not env.ref(f"l10n_id.{xmlid}", raise_if_not_found=False):
+            env['ir.model.data'].create({
+                "name": xmlid,
+                "module": "l10n_id",
+                "model": "account.tax.group",
+                "res_id": env['account.tax.group'].create({'name': name}).id,
+                'noupdate': True
+            })
+
+    # For all taxes linked to the tax_ST1 and tax_PT1, set thte tax group to non-luxury goods
+    # if no changes to amount and tax group yet
+    tax_group_id = env.ref("l10n_id.l10n_id_tax_group_non_luxury_goods", raise_if_not_found=False)
+    default_group = env['account.tax']._default_tax_group()
+    id_chart = env.ref("l10n_id.l10n_id_chart", raise_if_not_found=False)
+    companies = env['res.company'].search([('chart_template_id', 'child_of', id_chart.id)])
```

`id_chart.id` may raise in case `id_chart` is not found
We can return if the chart does not exist I think.